service/history/workflow/delete_manager.go

```diff
@@ -209,9 +206,21 @@ func (m *DeleteManagerImpl) deleteWorkflowExecutionInternal(
 		currentBranchToken = nil
 	}
 
-	completionEvent, err := ms.GetCompletionEvent(ctx)
-	if err != nil {
-		return err
+	var startTime *time.Time
+	var closeTime *time.Time
+
+	if ms.GetExecutionState().State == enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
```

Comment:
It should be:
```suggestion
	if ms.GetExecutionState().State != enumsspb.WORKFLOW_EXECUTION_STATE_RUNNING {
```
to support other completed states like `FAILED`.

Reply:
Both state or status work

Reply:
This align with the ms.getCompletionEvent

Reply:
revert delete manager changes